tests/gradients/test_parameter_shift.py

```diff
@@ -1297,7 +1297,7 @@ def __mul__(self, other):
                 return SpecialObject(self.val * other)
 
             def __add__(self, other):
-                newval = self.val + other.val if isinstance(other, self.__class__) else other
+                newval = self.val + (other.val if isinstance(other, self.__class__) else other)

```

Comment:
This was a silent bug: The addition previously was executed _before_ the Boolean clause, yielding `new=other` for the case of `not isinstance(other, self.__class__)`.